src/libs/dutil/shelutil.cpp

```diff
@@ -268,7 +268,22 @@ static HRESULT GetDesktopShellView(
     ExitOnFailure(hr, "Failed to get shell view.");
 
     hr = psw->FindWindowSW(&vEmpty, &vEmpty, SWC_DESKTOP, (long*)&hwnd, SWFO_NEEDDISPATCH, &pdisp);
-    ExitOnFailure(hr, "Failed to get desktop window.");
+    if (S_FALSE == hr)
+    {
+        //Windows XP
+        IShellFolder* folder = NULL;
```

Comment:
Probably need to release the `folder` and it should probably be called `pFolder` since it is a pointer.